modules/library/main/src/main/java/org/geotools/filter/function/UniqueIntervalFunction.java

```diff
@@ -137,4 +146,38 @@ public Object evaluate(Object feature) {
         }
         return calculate((SimpleFeatureCollection) feature);
     }
+
+    private double[] getPercentages(FeatureCollection collection, Set[] values) throws IOException {
+        Expression prop = getParameters().get(0);
+        GroupByVisitor groupBy =
+                new GroupByVisitor(Aggregate.COUNT, prop, Arrays.asList(prop), null);
+        collection.accepts(groupBy, null);
+        Object[] result = groupBy.getResult().toArray();
```

Comment:
Same observation as the other group by approach.

Reply:
fixed